crates/ra_hir_ty/src/traits/chalk/mapping.rs

```diff
@@ -30,7 +30,7 @@ impl ToChalk for Ty {
             Ty::Apply(apply_ty) => match apply_ty.ctor {
                 TypeCtor::Ref(m) => ref_to_chalk(db, m, apply_ty.parameters),
                 TypeCtor::Array => array_to_chalk(db, apply_ty.parameters),
-                TypeCtor::FnPtr { num_args: _ } => {
+                TypeCtor::FnPtr { num_args: _, is_varargs: _ } => {
```

Comment:
this might be a problem since function pointers will lose their varargness if they go through Chalk. should probably be fine for now though, but at least a FIXME would be good. Alternatively, you could do `is_varargs: false` here, then the varargs case would be handled by interning the type constructor (basically treating the type like a struct), which would mean that varargness would be preserved, but vararg function pointers wouldn't be treated as function pointers by Chalk.

Chalk issue tracking this: rust-lang/chalk#507

Reply:
Okay, FIXME added